components/tidb_query/src/rpn_expr/impl_string.rs

```diff
@@ -40,6 +40,25 @@ pub fn concat(args: &[&Option<Bytes>]) -> Result<Option<Bytes>> {
     Ok(Some(output))
 }
 
+#[rpn_fn(varg, min_args = 2)]
+#[inline]
+pub fn concat_ws(args: &[&Option<Bytes>]) -> Result<Option<Bytes>> {
+    if let Some(sep) = args[0] {
+        let mut output = Bytes::new();
+        let rest = &args[1..];
+        let iter = rest.iter().filter_map(|x| x.as_ref()).enumerate();
+        for (idx, x) in iter {
```

Comment:
You can use `join` to simplify the code.

Reply:
done.